### DISCUSSION TITLE
Does the current version of scalafx still work with scalafxml?

### DISCUSSION THREAD

**Discussion by kwalcock** [scalafx/scalafx#408] [Q&A] [answered]
I'm using Java 17, because it had been tested and approved, so build.sbt includes things like

```
ThisBuild / scalaVersion := "2.13.13"

  val javafx = Seq("base", "controls", "fxml", "graphics", "media", "swing", "web")
      .map(m => "org.openjfx" % s"javafx-$m" % "17" classifier osName)
```
and then
```
    "org.scalafx"            %% "scalafx"                 % "22.0.0-R33",
    "org.scalafx"            %% "scalafxml-core-sfx8"     % "0.5",
```

The main program includes
```
object ScalaFXHelloWorld extends JFXApp3 {

  override def start(): Unit = {
    val resource = getClass.getResource("/scenebuilder.fxml")
    val root = FXMLView(resource, NoDependencyResolver)

    stage = new JFXApp3.PrimaryStage {
      title = "Title"
      scene = new Scene(root)
    }
```

but my controller variables aren't getting initialized properly in
```
import javafx.fxml.{FXML => JFXML, Initializable => JInitializable}
import scalafx.scene.control.ListView

trait DatabaseInterface extends JInitializable {
  def initialize(url: URL, rb: util.ResourceBundle): Unit = {
    println("Initializing...")
  }
}

@sfxml
class DatabaseController(lvRegion: ListView[String], lvDataset: ListView[String]) extends DatabaseInterface {
  // more code to do something with lvRegion
```

lvRegion and lvDataset are null when more code is executed and initialize.

There are all sorts of nearly conflicting examples with and without @sfxml, @FXML, the controller extending a trait, the controller extending Initializable, variables being initialized in initialize(...), maybe with other dependency resolvers, etc.  I have not found a combination that works.  Most examples do run, but with very outdated code or non-obvious settings in sbt or no use of fxml so that I can't use them directly.  Is the project just obsolete or can someone spot an issue with my implementation?

**Comment by jpsacha** [comment]
ScalaFX 22.0.0-R33 works with ScalaFXML. Keep in mind that ScalaFXML currently works only with Scala 2.13 and 2.12, it does not work with Scala 3. Future versions of Scala 3 may bring macro annotations back, so  ScalaFXML may be ported to Scala 3.

If you found "conflicting" examples using FXML it may be due to examples being for different Scala versions. Scala 3 does not use `@FXML`. 

If your controls are not being initialized it may be due to the FXML file not being loaded (wrong path).  You can find a complete example here:
https://github.com/scalafx/ScalaFX-Tutorials/tree/master/scalafxml-example

**Comment by kwalcock** [comment]
Thanks for convincing me to keep trying.  The bulk of the problem seems to be that the SceneBuilder I used to edit the fxml files uses id="" while AdoptionForm.fxml uses fx:id="".   If I manually edit my IDs, it works as hoped.  So far, I do not see fx:id displayed anywhere in SceneBuilder (version 8 or 21) if I open AdoptionForm.fxml, and I can't get a SceneBuilder to create an fx:id.  I see in AdoptionForm.fxml
```
xmlns="http://javafx.com/javafx/2.2"
```
so perhaps it was created with a very old version that is required for compatibility with sfxml.  I hope that's not the case.

**Comment by kwalcock** [comment, marked as answer]
I tracked it down.  It was right there in the code pane.  Too many IDs.  Newbie mistake.  Thanks.
